core/src/identity/ed25519.rs

```diff
@@ -178,6 +191,18 @@ impl SecretKey {
         sk_bytes.zeroize();
         Ok(SecretKey(secret))
     }
+
+    /// View the secret key as bytes.
+    pub fn encode(&self) -> &[u8; 32] {
+        self.0.as_bytes()
+    }
+
+    /// Decode a private key from a byte array as produced by `encode`.
+    pub fn decode(k: &[u8]) -> Result<SecretKey, DecodingError> {
```

Comment:
How does `decode` differ from `from_bytes` other than not zeroizing the input? Why duplicate the functionality?

Reply:
Pardon my confusion. I now see that this PR is unnecessary.